File: solutions/design-hashset/design-hashset.js
/**
 * Initialize your data structure here.
 */
var MyHashSet = function() {
    this.storage = {};
};

/** 
 * @param {number} key
 * @return {void}
 */
MyHashSet.prototype.add = function(key) {
    if (!this.contains(key)) {
        this.storage[key] = 1;
    }
};

/** 
 * @param {number} key
 * @return {void}
 */
MyHashSet.prototype.remove = function(key) {
    if (this.contains(key)) {
        delete this.storage[key];
    }
};

/**
 * Returns true if this set contains the specified element 
 * @param {number} key
 * @return {boolean}
 */
MyHashSet.prototype.contains = function(key) {
    if (this.storage[key]) {
        return true;
    }
    
    return false;
};

/** 
 * Your MyHashSet object will be instantiated and called as such:
 * var obj = new MyHashSet()
 * obj.add(key)
 * obj.remove(key)
 * var param_3 = obj.contains(key)
 */

File: solutions/design-hashset/design-hashset.py
class MyHashSet:

    def __init__(self):
        self.storage = {}
        

    def add(self, key: int) -> None:
        if not self.contains(key):
            self.storage[key] = key

    def remove(self, key: int) -> None:
        if self.contains(key):
            del self.storage[key]

    def contains(self, key: int) -> bool:
        if key in self.storage:
            return True
        
        return False
        


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)
